src/libs/actions/Card.ts

```diff
@@ -0,0 +1,184 @@
+                key: ONYXKEYS.FORMS.REPORT_PHYSICAL_CARD_FORM,
+                value: {
+                    isLoading: true,
+                    errors: null,
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.FORMS.REPORT_PHYSICAL_CARD_FORM,
+                value: {
+                    isLoading: false,
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.FORMS.REPORT_PHYSICAL_CARD_FORM,
+                value: {
+                    isLoading: false,
+                },
+            },
+        ],
+    });
+}
+
+/**
+ * Activates the physical Expensify card based on the last four digits of the card number
+ */
+function activatePhysicalExpensifyCard(cardLastFourDigits: string, cardID: number) {
+    type ActivatePhysicalExpensifyCardParams = {
+        cardLastFourDigits: string;
+        cardID: number;
+    };
+
+    const parameters: ActivatePhysicalExpensifyCardParams = {
+        cardLastFourDigits,
+        cardID,
+    };
+
+    API.write('ActivatePhysicalExpensifyCard', parameters, {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.CARD_LIST,
+                value: {
+                    [cardID]: {
+                        errors: null,
+                        isLoading: true,
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.CARD_LIST,
+                value: {
+                    [cardID]: {
+                        isLoading: false,
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.CARD_LIST,
+                value: {
+                    [cardID]: {
+                        isLoading: false,
+                    },
+                },
+            },
+        ],
+    });
+}
+
+/**
+ * Clears errors for a specific cardID
+ */
+function clearCardListErrors(cardID: number) {
+    Onyx.merge(ONYXKEYS.CARD_LIST, {[cardID]: {errors: null, isLoading: false}});
+}
+
+/**
+ * Makes an API call to get virtual card details (pan, cvv, expiration date, address)
+ * This function purposefully uses `makeRequestWithSideEffects` method. For security reason
+ * card details cannot be persisted in Onyx and have to be asked for each time a user want's to
+ * reveal them.
+ *
+ * @param cardID - virtual card ID
+ *
+ * @returns promise with card details object
+ */
+function revealVirtualCardDetails(cardID: number): Promise<Response> {
+    return new Promise((resolve, reject) => {
+        type RevealExpensifyCardDetailsParams = {cardID: number};
```

Comment:
I noticed types in this file are inside the functions, would it make sense to keep all of them on top of the file or in a separate `types.ts` file for better readability?

Reply:
@TMisiukiewicz That's the specific pattern we agreed to follow for API parameters! We were asked to create  a separate type for the parameters and keep it closer to the API call